src/pipecat/services/doubao.py

```diff
@@ -0,0 +1,116 @@
+
+# doubao is a Chinese llm service belonging to Bytedance
+# install doubao sdk
+# pip install 'volcengine-python-sdk[ark]'
+
+import os
+from dotenv import load_dotenv
+from volcenginesdkarkruntime import Ark
+from typing import AsyncGenerator
+from loguru import logger
+import json
+from pipecat.services.ai_services import LLMService
+from pipecat.frames.frames import Frame, LLMMessagesFrame, TextFrame, LLMFullResponseStartFrame, LLMFullResponseEndFrame, UserStartedSpeakingFrame
+from pipecat.processors.aggregators.openai_llm_context import OpenAILLMContext, OpenAILLMContextFrame
+
+load_dotenv()
+
+
+class DoubaoLLMService(LLMService):
+    def __init__(self, *, model: str, **kwargs):
+        super().__init__(model=model, **kwargs)
+        self._client = Ark()
+        self._model: str = model
+        # detect if user start a new conversation during llm generation
+        self.llm_user_interrupt = False
+        # self._enable_metrics = True
+        # self._report_only_initial_ttfb = True
+
+    async def get_chat_completions(
+            self,
+            messages: list) -> AsyncGenerator[dict, None]:
+        try:
+            stream = self._client.chat.completions.create(
+                model=self._model,
+                messages=messages,
+                stream=True
+            )
+            for chunk in stream:
+                if not chunk.choices:
+                    continue
+                char = chunk.choices[0].delta.content
+                yield {"choices": [{"delta": {"content": char}}]}
+        except Exception as e:
+            logger.error(f"Error in doubao API call: {e}")
+            yield {"choices": [{"delta": {"content": f"Error: {str(e)}"}}]}
+
+    def _convert_context_to_doubao_format(self, context: OpenAILLMContext) -> list:
+        doubao_messages = [
+            {
+                "role": "system",
+                "content": "You are a helpful assistant."
+            }
+        ]
```

Comment:
why do we need this? shouldn't the initial context already have an initial prompt?